wallet.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import hathorLib from '@hathor/wallet-lib';
+import MemoryStore from './store';
+
+import EventEmitter from 'events';
+
+/**
+ * This is a Wallet that is supposed to be simple to be used by a third-party app.
+ *
+ * This class handles all the details of syncing, including receiving the same transaction
+ * multiple times from the server. It also keeps the balance of the tokens updated.
+ *
+ * It has the following states:
+ * - CLOSED: When it is disconnected from the server.
+ * - CONNECTING: When it is connecting to the server.
+ * - SYNCING: When it has connected and is syncing the transaction history.
+ * - READY: When it is ready to be used.
+ *
+ * You can subscribe for the following events:
+ * - state: When the state of the Wallet changes.
+ * - new-tx: When a new tx arrives.
+ * - update-tx: When a known tx is updated. Usually, it happens when one of its outputs is spent.
+ **/
+class Wallet extends EventEmitter {
+  constructor({ server, seed }) {
+    super();
+
+    this.state = this.CLOSED;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
+
+    this.server = server;
+    this.seed = seed;
+
+    this.passphrase = '';
+    this.pinCode = '123';
+    this.password = '123';
+  }
+
+  /**
+   * Called when loading the history of transactions.
+   * It is called every HTTP Request to get the history of a set of addresses.
+   * Usually, this is called multiple times.
+   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
+   *
+   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
+   **/
+  onAddressesLoaded(result) {
+    //console.log('result', result);
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    if (value) {
+      this.setState(Wallet.SYNCING);
+      hathorLib.wallet.loadAddressHistory(0, hathorLib.constants.GAP_LIMIT).then(() => {
+        this.setState(Wallet.READY);
+      }).catch((error) => {
+        throw error;
+      })
+    }
+  }
+
+  getAddressToUse() {
+    return hathorLib.wallet.getAddressToUse();
+  }
+
+  getCurrentAddress() {
+    return hathorLib.wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      this.onNewTx(wsData);
+    }
+  }
+
+  reloadData() {
```

Comment:
We can remove this if we're not using

Reply:
We will use it soon. I'll leave it right now.